app/src/main/java/ch/sdp/vibester/fragment/ScoreBoardFragment.kt

```diff
@@ -98,17 +104,41 @@ class ScoreBoardActivity : AppCompatActivity(), OnItemClickListener {
 
     private fun showPlayersPosition(players: MutableList<User>?) {
         userScoreboardAdapter = UserScoreboardAdapter(players!!, genre, this)
-        findViewById<RecyclerView>(R.id.recycler_view)!!.adapter = userScoreboardAdapter
+        requireView().findViewById<RecyclerView>(R.id.recycler_view)!!.adapter = userScoreboardAdapter
     }
 
     /**
      * @param position index of the item on click
      * go to the profile of the player at index position
      */
     override fun onItemClick(position: Int) {
-        val intent = Intent(this, PublicProfileActivity::class.java)
+        val intent = Intent(requireActivity(), PublicProfileActivity::class.java)
         intent.putExtra("UserId", players?.get(position)?.uid)
         intent.putExtra("ScoresOrFollowing", R.string.profile_scores.toString() )
         startActivity(intent)
     }
+
+    override fun onClick(v: View?) {
+            when(v!!.getId()) {
+                R.id.btsButton -> {
+                    genre = "BTS"
+                }
+                R.id.kpopButton -> {
+                    genre = "kpop"
+                }
+                R.id.imagDragonsButton -> {
+                    genre = "Imagine Dragons"
+                }
+                R.id.billieEilishButton -> {
+                    genre = "Billie Eilish"
+                }
+                R.id.rockButton -> {
+                    genre = "rock"
+                }
+                R.id.topTracksButton -> {
+                    genre = "top tracks"
+                }
+            }
+            selectScoreboard()
+    }
```

Comment:
Good idea to use a switch case instead of what was previously in selectScoreboard 👍